packages/server/src/routes/agencies.js

```diff
@@ -53,6 +54,28 @@ router.delete('/del/:agency', requireAdminUser, async (req, res) => {
     res.json(result);
 });
 
+router.put('/email_subscription/:agency', requireAdminUser, async (req, res) => {
+    const { agency } = req.params;
+    const { user } = req.session;
+
+    const allowed = await isUserAuthorized(user, agency);
+
+    if (!allowed) {
+        res.sendStatus(403);
+        return;
+    }
+
+    const { preferences } = req.body;
+
+    try {
+        await setAgencyEmailSubscriptionPreference(agency, preferences);
+        res.sendStatus(200);
+    } catch (e) {
+        console.error(`Unable to update agency email preferences for agency: ${agency} preferences: ${preferences}`);
```

Comment:
To help developers diagnose errors, I like to include the error (e.g.` e.toString()` I think) as part of the console log. I also like to include it in the page returned to the user. I realize this may be confusing to the end user, but it does help diagnose the problem.

Reply:
good idea! I'll add that in 👍

Reply:
Done 👍